torch/_inductor/utils.py

```diff
@@ -459,7 +459,9 @@ def has_incompatible_cudagraph_ops(gm):
 
 
 instance_descriptor = collections.namedtuple(
-    "instance_descriptor", ["divisible_by_16", "equal_to_1"]
+    "instance_descriptor",
+    ["divisible_by_16", "equal_to_1", "ids_of_folded_args", "divisible_by_8"],
+    defaults=[set(), set(), set(), set()],
```

Comment:
1)
mutable defaults is a bad idea because if I do:
```
a=instance_descriptor()
b=instance_descriptor()
```
then `a.ids_of_folded_args` will alias `b.ids_of_folded_args` so if I mutate one the other will change.

2) sets are non-determinstic, perhaps tuples?

3 We should also populate divisibile_by_8 the same way we do divisible_by_16

Reply:
I literally copy the definition from triton source code, lol. We need to be consistent with how triton interpret each field. E.g., if triton expect a set but we provide a tuple, a python failure may happen.

How about we change set() to frozenset() to avoid 1st item you mentioned and also make sure the type is still a set?


For 'divisible_by_8' , the way triton set it is here: https://github.com/openai/triton/blob/main/python/triton/runtime/jit.py#L208-L222 . Not sure if it's by design that for a tensor input it will always return false. I think I'll probably copy the logic. Or maybe @Jokeren / @Philippe Tillet could shed some light here.

Reply:
`divisible_by_8` is by so far only used for TMA. See comment here https://github.com/openai/triton/blob/5282ed890d453e10b9ee30076ef89115dd197761/python/triton/runtime/jit.py#L157

Reply:
@jansel  I'll change default value to tuple as you suggested since looking into triton's code, there are usage like [this](https://github.com/openai/triton/blob/main/python/triton/runtime/jit.py#L235-L236) which also initialize `instance_descriptor` with tuples.